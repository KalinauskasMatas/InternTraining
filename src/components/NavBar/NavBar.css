nav > ul {
  background-color: var(--dark-button-color);
  color: var(--bright-text-color);
  display: flex;
  justify-content: space-between;
  margin: 10px 47px 0px 43px;
  padding: 9px 10px 10px 21px;
  border-radius: 25px;
}

nav > ul li {
  display: inline-block;
  list-style: none;
  font-size: 16px;
  font-weight: 700;
  padding: 6px 8px 6px 8px;
  margin: 0 12px 0 0;
  letter-spacing: 0.2px;
  cursor: pointer;
}

#logout {
  text-align: end;
}

nav > ul .left li:first-child{
  font-size: 20px;
}

nav .profile-link {
  margin-left: -11px;
}

nav a {
  color: var(--bright-text-color);
  text-decoration: none;
}

nav a:visited {
  color: var(--bright-text-color);
}