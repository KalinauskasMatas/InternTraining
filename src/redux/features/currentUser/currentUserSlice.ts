import { createSlice, PayloadAction } from "@reduxjs/toolkit";
import { RootState } from "../../store";
import { UserState } from "../../interfaces";

const initialState: UserState =
  JSON.parse(localStorage.getItem("currUser") as string) || {};

const currentUserSlice = createSlice({
  name: "currentUser",
  initialState,
  reducers: {
    setUser: (state, action: PayloadAction<UserState>) => {
      state.fname = action.payload.fname;
      state.surname = action.payload.surname;
      state.email = action.payload.email;
      state.password = action.payload.password;
      localStorage.setItem("currUser", JSON.stringify(state));
    },
    clearUser: (state) => {
      localStorage.removeItem("currUser");
      state = { ...initialState };
      console.log(state);
    },
  },
});

export const { setUser, clearUser } = currentUserSlice.actions;

export const selectCurrUser = (state: RootState) => state.currentUser;

export default currentUserSlice.reducer;
