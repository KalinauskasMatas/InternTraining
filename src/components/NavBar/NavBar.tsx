import React from "react";
import { clearUser } from "../../redux/features/currentUser/currentUserSlice";
import { useAppDispatch } from "../../redux/hooks";

import "./NavBar.css";

const NavBar = () => {
  const dispatch = useAppDispatch();

  const handleLogout = () => {
    dispatch(clearUser());
    window.location.reload();
  };

  return (
    <nav>
      <ul>
        <span className="left">
          <li>Home</li>
          <li>Your movies</li>
          <li className="profile-link">Profile</li>
        </span>
        <span>
          <li id="logout" onClick={handleLogout}>
            Logout
          </li>
        </span>
      </ul>
    </nav>
  );
};

export default NavBar;
