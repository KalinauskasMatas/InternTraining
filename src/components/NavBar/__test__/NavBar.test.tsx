import NavBar from "../NavBar";

import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";

import renderer from "react-test-renderer";

import { Provider } from "react-redux";
import { configureStore } from "@reduxjs/toolkit";

import { BrowserRouter } from "react-router-dom";
import currentUserSlice, {
  setCurrUser,
} from "../../../redux/features/currentUser/currentUserSlice";

const store = configureStore({
  reducer: {
    currentUser: currentUserSlice,
  },
});

afterEach(cleanup);

const testUser = {
  fname: "test",
  surname: "user",
  email: "test@user",
  password: "12345678",
  rentMovies: [],
  isAdmin: false,
};

it("renders without crashing not admin", () => {
  store.dispatch(setCurrUser(testUser));
  render(
    <Provider store={store}>
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>
    </Provider>
  );
});

it("renders without crashing admin", () => {
  store.dispatch(setCurrUser({ ...testUser, isAdmin: true }));
  render(
    <Provider store={store}>
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>
    </Provider>
  );
});

it("matches snapshot non admin", () => {
  store.dispatch(setCurrUser(testUser));
  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <NavBar />
        </BrowserRouter>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("matches snapshot admin", () => {
  store.dispatch(setCurrUser({ ...testUser, isAdmin: true }));
  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <NavBar />
        </BrowserRouter>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
