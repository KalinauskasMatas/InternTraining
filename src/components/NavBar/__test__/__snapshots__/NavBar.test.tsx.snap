// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot admin 1`] = `
<nav>
  <ul>
    <span
      className="left"
    >
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/yourMovies"
          onClick={[Function]}
        >
          Your movies
        </a>
      </li>
      <li
        className="profile-link"
      >
        <a
          href="/profile"
          onClick={[Function]}
        >
          Profile
        </a>
      </li>
    </span>
    <span>
      <li>
        <a
          href="/users"
          onClick={[Function]}
        >
          Users
        </a>
      </li>
      <li
        id="logout"
        onClick={[Function]}
      >
        Logout
      </li>
    </span>
  </ul>
</nav>
`;

exports[`matches snapshot non admin 1`] = `
<nav>
  <ul>
    <span
      className="left"
    >
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/yourMovies"
          onClick={[Function]}
        >
          Your movies
        </a>
      </li>
      <li
        className="profile-link"
      >
        <a
          href="/profile"
          onClick={[Function]}
        >
          Profile
        </a>
      </li>
    </span>
    <span>
      <li
        id="logout"
        onClick={[Function]}
      >
        Logout
      </li>
    </span>
  </ul>
</nav>
`;
